tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/AlfrescoStackInitializer.java

```diff
@@ -297,4 +282,115 @@ protected GenericContainer createAlfrescoContainer()
                         BindMode.READ_ONLY);
     }
+
+    public static ImagesConfig getImagesConfig()
+    {
+        return DefaultImagesConfig.INSTANCE;
+    }
+
+    public interface ImagesConfig
+    {
+        String getReIndexingImage();
+
+        String getLiveIndexingImage();
+
+        String getElasticsearchImage();
+
+        String getActiveMqImage();
+
+        String getTransformRouterImage();
+
+        String getTransformCoreAIOImage();
+
+        String getSharedFileStoreImage();
+
+        String getPostgreSQLImage();
+
+        String getRepositoryImage();
+
+        String getKibanaImage();
+    }
+
+    private static final class DefaultImagesConfig implements ImagesConfig
+    {
+        private static final DefaultImagesConfig INSTANCE = new DefaultImagesConfig(EnvHelper::getEnvProperty, MavenPropertyHelper::getMavenProperty);
+        private final Function<String, String> envProperties;
+        private final Function<String, String> mavenProperties;
+
+        DefaultImagesConfig(Function<String, String> envProperties, Function<String, String> mavenProperties)
+        {
+            this.envProperties = envProperties;
+            this.mavenProperties = mavenProperties;
+        }
+
+        @Override
+        public String getReIndexingImage()
+        {
+            return "quay.io/alfresco/alfresco-elasticsearch-reindexing:" + getElasticsearchConnectorImageTag();
+        }
+
+        @Override
+        public String getLiveIndexingImage()
+        {
+            return "quay.io/alfresco/alfresco-elasticsearch-live-indexing:" + getElasticsearchConnectorImageTag();
+        }
+
+        @Override
+        public String getElasticsearchImage()
+        {
+            return "docker.elastic.co/elasticsearch/elasticsearch:" + envProperties.apply("ES_TAG");
+        }
+
+        @Override
+        public String getActiveMqImage()
+        {
+            return "alfresco/alfresco-activemq:" + envProperties.apply("ACTIVEMQ_TAG");
+        }
+
+        @Override
+        public String getTransformRouterImage()
+        {
+            return "quay.io/alfresco/alfresco-transform-router:" + mavenProperties.apply("dependency.alfresco-transform-service.version");
+        }
+
+        @Override
+        public String getTransformCoreAIOImage()
+        {
+            return "alfresco/alfresco-transform-core-aio:" + mavenProperties.apply("dependency.alfresco-transform-core.version");
+        }
+
+        @Override
+        public String getSharedFileStoreImage()
+        {
+            return "quay.io/alfresco/alfresco-shared-file-store:" + mavenProperties.apply("dependency.alfresco-transform-service.version");
+        }
+
+        @Override
+        public String getPostgreSQLImage()
+        {
+            return "postgres:" + envProperties.apply("POSTGRES_TAG");
+        }
+
+        @Override
+        public String getRepositoryImage()
+        {
+            return "alfresco/alfresco-content-repository:latest";
```

Comment:
All others seem to get a specific version except this one, it uses "latest", why?

Reply:
Nevermind, I started reviewing from the wrong angle, now I see it's a refactor, 👍